src/app/devices/components/DeviceHeaderForms/AddDeviceForm/AddDeviceInterfaces/AddDeviceInterfaces.tsx

```diff
@@ -49,59 +132,15 @@ export const AddDeviceInterfaces = (): JSX.Element => {
   return (
     <>
       <h4>Interfaces</h4>
-      {interfaces.map((iface, i) => {
-        const showSubnetField =
-          iface.ip_assignment === DeviceIpAssignment.STATIC;
-        const showIpAddressField =
-          iface.ip_assignment === DeviceIpAssignment.STATIC ||
-          iface.ip_assignment === DeviceIpAssignment.EXTERNAL;
-        const deleteDisabled = interfaces.length <= 1;
-
-        return (
-          <Card data-testid="interface-card" key={iface.id}>
-            <FormikField
-              label="Name"
-              name={`interfaces[${i}].name`}
-              type="text"
-            />
-            <MacAddressField
-              label="MAC address"
-              name={`interfaces[${i}].mac`}
-              required
-            />
-            <IpAssignmentSelect
-              name={`interfaces[${i}].ip_assignment`}
-              onChange={(e: ChangeEvent<HTMLSelectElement>) => {
-                handleChange(e);
-                setFieldValue(`interfaces[${i}].subnet`, "");
-                setFieldValue(`interfaces[${i}].ip_address`, "");
-              }}
-              required
-            />
-            {showSubnetField ? (
-              <SubnetSelect
-                data-testid="subnet-field"
-                name={`interfaces[${i}].subnet`}
-              />
-            ) : null}
-            {showIpAddressField ? (
-              <FormikField
-                data-testid="ip-address-field"
-                label="IP address"
-                name={`interfaces[${i}].ip_address`}
-                type="text"
-              />
-            ) : null}
-            {!deleteDisabled ? (
-              <div className="u-align--right">
-                <Button onClick={() => removeInterface(iface.id)} type="button">
-                  Delete
-                </Button>
-              </div>
-            ) : null}
-          </Card>
-        );
-      })}
+      {interfaces.map((iface, i) => (
+        <AddDeviceInterfaceFields
+          deleteDisabled={interfaces.length === 1}
+          iface={iface}
+          index={i}
```

Comment:
Using array indices as React keys or in dynamic form field names can lead to unexpected behavior.

https://react.dev/learn/rendering-lists#rules-of-keys

It's best to avoid this pattern. I'd suggest to use a unique, stable identifier for each interface (e.g., iface.id) instead.

```suggestion
```